#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 5.272566532318279e-05 -0.9121910998466397 -0.40976504802213437 3.1566902004042756
  position 0.3241810857456441 23.217463737062545 -19.103373720769707
}
TexturedBackground {
}
TexturedBackgroundLight {
}
UnevenTerrain {
  name "virtual garden"
  size 10 0 10
  appearance SandyGround {
    colorOverride 0 0.517906 0.133974
    textureTransform TextureTransform {
      scale 50 50
    }
    IBLStrength 3
  }
}
PicketFence {
  translation 4.2 0 -4.2
  rotation 0 1 0 3.14159
  name "front fence"
  numberOfSegments 5
}
PicketFence {
  translation -4.2 0 4.2
  name "back fence"
  numberOfSegments 5
}
PicketFence {
  translation 4.2 0 4.2
  rotation 0 1 0 1.57
  name "left fence"
  numberOfSegments 5
}
PicketFence {
  translation -4.2 0 4.2
  rotation 0 1 0 1.5708
  name "right fence"
  numberOfSegments 5
}
PalmTree {
  translation -3 0 3
}
Oak {
  translation 3 0 -3
  name "oak tree 0"
}
Oak {
  translation 3 0 3
  name "oak tree 1"
}
Oak {
  translation -3 0 -3
  name "oak tree 2"
}
Wheelbarrow {
  translation 3 0 -0.5
  rotation 0 1 0 1.57
}
DogHouse {
  translation 2.5 0 1
  rotation 0 1 0 1.5708
}
TableWithUmbrella {
  translation 0 0 2.5
}
SwingCouch {
  translation -2.5 0 0
  rotation 0 1 0 3.14159
}
PottedTree {
  translation 1.5 0 -3
  rotation 0 0.9999999999999999 0 0
  name "potted tree 1"
}
PottedTree {
  translation -1.5 0 -3
  name "potted tree 2"
}
Apple {
  translation 0.65 0.04 -0.31
  name "apple 3"
}
Apple {
  translation -0.831999 0.0842798 1.16
  rotation 6.1248568571585574e-15 4.474980960000451e-13 -1 0.08765091923114961
  name "apple 2"
}
Apple {
  translation -0.895267 0.0496782 -1.91
  rotation 6.1248568571585574e-15 4.474980960000451e-13 -1 0.08765091923114961
  name "apple 6"
}
Apple {
  translation -0.894404 0.059646 -0.42
  rotation 6.1248568571585574e-15 4.474980960000451e-13 -1 0.08765091923114961
  name "apple 4"
}
Apple {
  translation 0.71992 0.038248 1.24
  rotation 6.1248568571585574e-15 4.474980960000451e-13 -1 0.08765091923114961
  name "apple 1"
}
Apple {
  translation 0.610357 0.0478777 -1.94
  rotation 6.1248568571585574e-15 4.474980960000451e-13 -1 0.08765091923114961
  name "apple 5"
}
Robot {
  translation 0.119183 0.1 0.181722
  rotation 0 1 0 -5.307179586466759e-06
  scale 1.5 1.5 1.5
  children [
    DEF base_link Shape {
      appearance PBRAppearance {
        baseColor 0.0659342 0.501793 0.00683604
        metalness 0
      }
      geometry Box {
        size 0.38 0.03 0.15
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.1 0 0.08
      }
      device [
        RotationalMotor {
          name "right_front_wheel"
        }
      ]
      endPoint Solid {
        translation 0.1 0 0.08
        rotation 1 1.099502589908099e-15 -3.926794963957497e-16 1.5699999999999963
        children [
          DEF wheel_shape Shape {
            appearance PBRAppearance {
              baseColor 0.8 0.5 0
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.055
            }
          }
        ]
        name "right_front_wheel"
        boundingObject USE wheel_shape
        physics DEF wheel_physics Physics {
        }
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.02 0 0.08
      }
      device [
        RotationalMotor {
          name "right_mid_wheel"
        }
      ]
      endPoint Solid {
        translation -0.02 0 0.08
        rotation 1 6.710831237232049e-20 -6.710831237232049e-20 1.5699999999999972
        children [
          USE wheel_shape
        ]
        name "right_mid_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.14 0 0.08
      }
      device [
        RotationalMotor {
          name "right_rear_wheel"
        }
      ]
      endPoint Solid {
        translation -0.14 0 0.08
        rotation 1 -1.570717985583e-16 -7.853589927915e-17 1.569999999999995
        children [
          USE wheel_shape
        ]
        name "right_rear_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.1 0 -0.08
      }
      device [
        RotationalMotor {
          name "left_front_wheel"
        }
      ]
      endPoint Solid {
        translation 0.09999999999999999 -2.516291789653337e-17 -0.08
        rotation 1 7.853589927915012e-17 3.5341154675617555e-15 1.5699999999999912
        children [
          USE wheel_shape
        ]
        name "left_front_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.02 0 -0.08
      }
      device [
        RotationalMotor {
          name "left_mid_wheel"
        }
      ]
      endPoint Solid {
        translation -0.02 0 -0.08
        rotation 1 -1.533904282795896e-19 3.643022671640254e-19 1.569999999999998
        children [
          USE wheel_shape
        ]
        name "left_mid_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.14 0 -0.08
      }
      device [
        RotationalMotor {
          name "left_rear_wheel"
        }
      ]
      endPoint Solid {
        translation -0.14 0 -0.08
        rotation 1 -5.497512949540489e-16 7.068230935123485e-16 1.5699999999999985
        children [
          USE wheel_shape
        ]
        name "left_rear_wheel"
        boundingObject USE wheel_shape
        physics USE wheel_physics
        locked TRUE
      }
    }
    DistanceSensor {
      translation 0.19 -0.0075 0.075
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_outer_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      locked TRUE
    }
    DistanceSensor {
      translation 0.19 -0.0075 0.025
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_mid_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      locked TRUE
    }
    DistanceSensor {
      translation 0.19 -0.0075 -0.075
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_outer_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      locked TRUE
    }
    DistanceSensor {
      translation 0.19 -0.0075 -0.025
      children [
        DEF ir_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_mid_sensor"
      boundingObject USE ir_shape
      physics DEF ir_physics Physics {
      }
      locked TRUE
    }
    DistanceSensor {
      translation 0.17 -0.0075 0.0749601
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right_side_sensor"
      boundingObject USE ir_shape
      physics USE ir_physics
      locked TRUE
    }
    DistanceSensor {
      translation 0.17 -0.0075 -0.0749601
      rotation 0 1 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left_side_sensor"
      boundingObject USE ir_shape
      physics USE ir_physics
      locked TRUE
    }
    Camera {
      translation 0.13 0.18 -1.16711e-09
      rotation 0 1 0 -1.57
      children [
        DEF camera_shape Shape {
          appearance PBRAppearance {
            baseColor 1 0.999969 0.999985
            metalness 0
          }
          geometry Cylinder {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "front_camera"
      boundingObject USE camera_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0 0.05 -0.065
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.37 0.1 0.01
          }
        }
      ]
      name "left basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0 0.05 0.065
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.37 0.1 0.01
          }
        }
      ]
      name "right basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation -0.18 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.01
          }
        }
      ]
      name "back basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0.18 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.01
          }
        }
      ]
      name "front basket"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0.15 0.05 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.14 0.1 0.05
          }
        }
      ]
      name "motor_box"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
    Solid {
      translation 0.125 0.1 0
      rotation 0 1 0 1.57
      children [
        DEF basket_shape Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.02 0.2 0.02
          }
        }
      ]
      name "camera_bar"
      boundingObject USE basket_shape
      physics Physics {
      }
      locked TRUE
    }
  ]
  name "GardenRobot"
  boundingObject USE base_link
  physics Physics {
  }
  controller "wasd_controller_six"
}
